### DISCUSSION TITLE
OpenSSL problem on Windows 10

### DISCUSSION THREAD

**Discussion by BrianHanke** [AcademySoftwareFoundation/OpenRV#175] [Build Issues] [answered]
I'm trying to build RV with the OpenEXR fix for DWAA, but my formerly working build setup is failing. I keep getting this error, any ideas how to fix? Thanks!

nasm  -Ox -f win64 -DNEAR -g -o crypto\aes\aesni-sha1-x86_64.obj "crypto\aes\aesni-sha1-x86_64.asm"
         panic: crypto\aes\aesni-sha1-x86_64.asm: assertion cv8_state.source_files != NULL failed at output/codeview.c:515
    14>NMAKE : fatal error U1077: 'C:\msys64\usr\bin\nasm.EXE' : return code '0x3' [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
         Stop.
    14>NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2' [C:\OpenRV\_bu
       ild\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]

**Comment by rogernelson** [comment]
Did you perhaps update your version of nmake or nasm? I had something similar and a clean build fixed the problem.  Can you try doing a `rvclean` and then `mkrv` or `mkrvd` (if you want debug)?

**Comment by BrianHanke** [comment]
@rogernelson Thanks Roger. This is using my old RV-building VM so nothing should have changed. 

What I ended up doing was downloading the source from a forked repo that hasn't been updated and patching in my changes, including the updated OpenEXR version. Working fine now! 

Seems that something changed since the initial release and my old build setup doesn't work... Anyway, no problem for now, but may have to revisit when the OCIO 2 changes hit.

**Comment by naotakahoriguchi** [comment]
I had a similar error with latest revision as below.
with -
  qt-unified-windows-x64-4.6.0-online.exe (Qt 5.15.2)
  strawberry-perl-5.32.1.1-64bit.msi
  msys2-x86_64-20230526.exe

Any hint?


```
    23>CustomBuild:
         panic: crypto\bn\x86_64-mont.asm: assertion cv8_state.source_files != NULL failed at output/codeview.c:515
    23>NMAKE : fatal error U1077: 'C:\msys64\usr\bin\nasm.EXE' : return code '0x3' [D:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
         Stop.
    23>NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2' [D:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
         Stop.
         Namespace(clean=False, configure=False, build=True, install=False, source=WindowsPath('D:/OpenRV/_build/RV_DEPS_OPENSSL/src'), output=WindowsPath('D:/OpenRV/_build/RV_DEPS_OPENSSL/install'), arch='', perlroot='C:/Strawberry/perl/bin'
         )
         Executing ['nmake'] from D:/OpenRV/_build/RV_DEPS_OPENSSL/src
         Traceback (most recent call last):
           File "D:\OpenRV\src\build\make_openssl.py", line 266, in <module>
             build()
           File "D:\OpenRV\src\build\make_openssl.py", line 210, in build
             subprocess.run(build_args, cwd=SOURCE_DIR).check_returncode()
           File "C:/msys64/mingw64/lib/python3.10/subprocess.py", line 457, in check_returncode
             raise CalledProcessError(self.returncode, self.args, self.stdout,
         subprocess.CalledProcessError: Command '['nmake']' returned non-zero exit status 2.
    23>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: 'D:\OpenRV\_build\CMakeFiles\275c3d1d115aa76c868d2701f2537e2f\RV_DEPS_OPENSSL-build.rule;D:\O
       penRV\_build\CMakeFiles\275c3d1d115aa76c868d2701f2537e2f\RV_DEPS_OPENSSL-install.rule;D:\OpenRV\_build\CMakeFiles\f006731585e5875045532d1ba81748cf\RV_DEPS_OPENSSL-complete.rule;D:\OpenRV\_build\CMakeFiles\a8ed0a6c6e555e200939fa450a4074
       d3\RV_DEPS_OPENSSL.rule' Custom build exited with code 1 [D:\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]

```

**Comment by paulbarton90** [comment, marked as answer]
In trying to get a more up to date build in Windows I found that this commit [624589f](https://github.com/AcademySoftwareFoundation/OpenRV/commit/624589f56fc870c4d8de83911ab43f53e9a9f36c) was causing the issue. More specifically it's the removal of `configure_args.append("no-asm")`. Once you add that line back in, I was able to successfully get a build within a Windows 10 virtual machine.

**Reply by BrianHanke** [reply]
This helps a lot, thanks! OpenSSL issue resolved. I'm still getting other errors as a result of whatever was changed in the build process since Open RV was first released, but one step at a time...
